src/context/auth/authContext.js

```diff
@@ -0,0 +1,68 @@
+import { createContext, useContext, useReducer } from "react";
+import { loginApi, signupApi } from "../../services/auth_services";
+import { reducer } from "./authReducer";
+
+const AuthContext = createContext();
+const AuthContextProvider = ({ children }) => {
+  const user = JSON.parse(localStorage.getItem("user"));
+  const intialState = user ?? {
+    isUserloggedIn: false,
+    token: "",
+    user: null,
+  };
+
+  const [{ isUserloggedIn, token }, dispatch] = useReducer(
+    reducer,
+    intialState
+  );
+  const signup = async ({ email, password, firstName, lastName }) => {
```

Comment:
I think this function should not be here, but i could be wrong

Reply:
I don't know . I think so it should be here